src/AppBundle/Entity/Report/ReportSubmission.php

```diff
@@ -78,13 +91,19 @@ class ReportSubmission
     /**
      * ReportSubmission constructor.
      *
-     * @param Report $report
+     * @param Report\Odr $report
      * @param User   $createdBy
      */
-    public function __construct(Report $report, User $createdBy)
+    public function __construct($report, User $createdBy)
     {
-        $this->report = $report;
-        $this->report->addReportSubmission($this);// double-link for UNIT test purposes
+        if ($report instanceof Report) {
+            $this->report = $report;
+            $this->report->addReportSubmission($this);// double-link for UNIT test purposes
+        } else if ($report instanceof Odr) {
+            $this->ndr = $report;
+        } else {
+            throw new \InvalidArgumentException(__METHOD__.' first argumnt should be a Report or an Ndr');
+        }
```

Comment:
Nothing to change really as its only a minor comment but using a ReportInterface instead of an AbstractReport in the constructor signature would mean we can get rid of all this checking and replace the signature with: 
__construct(ReportInterface $report, User $createdBy).

Reply:
changed AbstractReport into ReportInterface